multichain-testing/scripts/ymax-tool.ts

```diff
@@ -230,29 +347,127 @@ const main = async (
+    trace('getting instance', contract);
+    const { [contract]: instance } = fromEntries(
+      await walletKit.readPublished('agoricNames.instance'),
+    );
+    const result = await walletStore.saveOfferResult(
+      { instance, description },
+      description.replace(/^deliver /, ''),
+    );
+    trace('redeem result', result);
+    return;
+  }
+
+  const yc = walletStore.get<ContractControl<YMaxStartFn>>('ymaxControl');
+
+  if (values.getCreatorFacet) {
+    await walletStore.savingResult('creatorFacet', () => yc.getCreatorFacet());
+    return;
+  }
+
+  if (values.terminate) {
+    const { terminate: message } = values;
+    await yc.terminate({ message });
+    return;
+  }
+
+  if (values.installAndStart) {
+    const { installAndStart: bundleId } = values;
+    const { BLD, USDC } = fromEntries(
+      await walletKit.readPublished('agoricNames.issuer'),
+    );
+
+    // work around goofy devnet PoC token
+    const { upoc26 } = fromEntries(
+      await walletKit.readPublished('agoricNames.vbankAsset'),
+    );
+
+    await yc.installAndStart({
+      bundleId,
+      issuers: { USDC, BLD, Fee: BLD, Access: upoc26.issuer },
+      privateArgsOverrides: await overridesForEthChainInfo(walletKit),
+    });
+    return;
+  }
+
+  if (values.pruneStorage) {
+    console.error('finding outdated vstorage...');
+    const toPrune = await findOutdated(walletKit.vstorage);
+    console.log('toPrune', toPrune);
+    await yc.pruneChainStorage(toPrune);
+  }
+
+  if (values.invitePlanner) {
+    const { invitePlanner: planner } = values;
+    const cf =
+      walletStore.get<ZStarted<YMaxStartFn>['creatorFacet']>('creatorFacet');
+    const { postalService } = fromEntries(
+      await walletKit.readPublished('agoricNames.instance'),
+    );
+    await cf.deliverPlannerInvitation(planner, postalService);
+    return;
+  }
+
+  if (values['submit-for']) {
+    const portfolioId = Number(values['submit-for']);
+    const planner = walletStore.get<PortfolioPlanner>('planner');
+    await planner.submit(portfolioId, []);
+    return;
+  }
+
+  const positionData = parseTypedJSON(values.positions, GoalDataShape);
+  const targetAllocation = values['target-allocation']
+    ? parseTypedJSON(
+        values['target-allocation'],
+        TargetAllocationShape,
+        (_k, v) => (typeof v === 'number' ? BigInt(v) : v),
+        err => Error(`Invalid target allocation JSON: ${err.message}`),
+      )
+    : undefined;
+  console.debug({ positionData, targetAllocation });
   try {
-    // Pass the parsed options to openPosition
-    await openPosition(volume, {
-      address,
-      client,
-      walletKit,
-      now: Date.now,
-      skipPoll,
-      targetAllocationJson,
+    const opened = await openPositions(positionData, {
+      sig,
+      when: now(),
+      targetAllocation,
     });
+    trace('opened', opened);
+    const { path } = opened;
+    // XXX would be nice if readPublished allowed ^published.
```

Comment:
I suppose we could assume `published.published` will never exist and trim anything that starts with `published.` Sloppy though. I'd prefer a helper like `readPublished(rmp(existingpath))`